src/dmd/mars.d

```diff
@@ -2125,3 +2168,278 @@ private bool parseCommandLine(const ref Strings arguments, const size_t argc, re
     return errors;
 }
 
+
+private __gshared bool includeImports = false;
+// array of module patterns used to include/exclude imported modules
+private __gshared Array!(const(char)*) includeModulePatterns;
```

Comment:
It looks like this is only used internally (not exposed to the other GDC/LDC).  We want to move away from using `char*` for such things and use D's `string` facilities instead.  Would it be beneficial to use a `string[]` instead?

Reply:
Well they aren't strings, they are char*, their length is not precalculated.

Reply:
Also, it's likely that this code will be moved in the future and shared by LDC/GDC.

Reply:
That's fine, it's only at the interface boundaries between C++ and D that we need `char*`.  Even then it's been proposed to use `struct Slice(T) { size_t length; T* ptr; }`.  Just offering the idea; your call.